### ISSUE TITLE
SpeckleCircle needs X-axis and Y-axis properties for aligning plans

### ISSUE THREAD

**Issue by secretlyagoblin** [speckleworks/SpeckleCore#81]
#### Expected Behaviour

Start point on a circle domain should be based on the plane the curve has been defined in

#### Actual Behaviour

![create circle](https://user-images.githubusercontent.com/13952064/42188311-797a4e44-7ea7-11e8-8c00-2f06fc93737e.PNG)

**Comment by alvpickmans** [comment]
@secretlyagoblin as you suggest the [circle schema](https://github.com/speckleworks/SpeckleCore/blob/master/SpeckleCore/ModelObjectsExt.cs#L416-L425) (and arc by extension) seems to require some other property for better definition. I run into something similar while implementing arcs on the [dynamo converter](https://github.com/speckleworks/SpeckleDynamo/pull/18#issuecomment-394160060)

Rather than 2 vectors I thought more of having the start point as property as such would be slightly lighter and easier to re-construct on the target platform(?)

@didimitrie

**Comment by secretlyagoblin** [comment]
all the same to me! point + normal might be more useful for reducing error as they're further apart. Worth noting that Ellipse and Arc are derived from a xy axis.

**Comment by didimitrie** [comment]
More than happy to discuss these things! I've been running around in **circles** trying to fix these issues - pun intended. 

[SpeckleArcs](https://speckleworks.github.io/SpeckleSpecs/#tocSspecklearc) do have a plane property, which basically includes X, Y and Z (normal) dirs. I know there have been problems (https://github.com/speckleworks/SpeckleRhino/issues/170, https://github.com/speckleworks/SpeckleCore/issues/68). 

@alvpickmans: how are we doing for arcs from rhino/gh in dynamo at the moment? I remember I fixed this issue theoretically. 

@secretlyagoblin: would the plane be ok for circles? I guess this would give you the control you need.

@all: the big fuckup previously with arcs and rhino was the domain and how it's specified. Adding  this to arcs resolved https://github.com/speckleworks/SpeckleRhino/issues/170, but not really sure what's our status on this one.

**Comment by alvpickmans** [comment]
afaik although both Rhino and Dynamo have circle constructor based on [Plane and Radius](https://developer.rhino3d.com/api/RhinoCommon/html/M_Rhino_Geometry_Circle__ctor_2.htm), Dynamo only uses the plane's origin as its centre building the curve from the positive vector from the origin parallel to XY plane (rigth side if counterclockwise), while Rhino from the circle's plane X axis (starting points in blue on pics below).
The same applies to arcs, so in Rhino the start point is on the plane's X axis and start angle 0 while in Dynamo can be any.

Quickly checking [three.js api](https://threejs.org/docs/#api/geometries/CircleGeometry) , it considers the starting point same as Dynamo. It would be good to know how other possible clients handle this (Unity, Blender...), and define SpeckleCircle/Arc schemas according to what would be easier to implement.

![RhinoCircle](https://user-images.githubusercontent.com/12413808/42228910-0db55e70-7edd-11e8-958d-698cae8254f2.png)

![DynamoCircle](https://user-images.githubusercontent.com/12413808/42229399-52c498d6-7ede-11e8-8683-a681a64b0611.png)

**Comment by didimitrie** [comment]
then ping @secretlyagoblin and  @cwmorse regarding unity circles and @tsvilans for blender circles. 

It might also be the case that this level of control is not needed in Blender/Unity/Threejs, and we need to nail  it down for now Dynamo - Gh/Rh. 

@alvpickmans, talk more irl tomorrow, but we can have start points added too to circles and arcs.

**Comment by secretlyagoblin** [comment]
@didimitrie yep the plane should be fine, thanks! I'm using verbnurbs and it considers a circle to be a special case arc, which is probably a good way of thinking about it.

@alvpickmans that almost feels like bug on the part of Dynamo! I can't imagine that being preferred behaviour.

**Comment by alvpickmans** [comment]
@secretlyagoblin I agree on treating circles as "closed arcs". I wouldn't say it is a bug, probably just a different interpretation of the geometry, just like nurbcurves are defined differently (great old post [here](http://www.theprovingground.org/2012/09/interoperable-geometry-part-1-curves.html)).
In any case, there is no harm in asking Dynamo/Autodesk :)

**Comment by didimitrie** [comment]
I got in this when i reduced lines to a polyline with only two point coords in its array. Had to bring lines back because of public outcry. 

So would keep arcs as arcs on the speckle side for now...

**Comment by didimitrie** [comment]
![image](https://user-images.githubusercontent.com/7696515/42288897-a9d093c4-7fb4-11e8-81d1-46e90f3ca09f.png)

okay this is sorted now! I've remade a quick def to spin up some circles. will be closed soon...

**Comment by alvpickmans** [comment]
Made it work in Dynamo as well! It was indeed a bug on its geometry library, but it is resolved until they release the fix with version 2.1.
This can be closed now.
![image](https://user-images.githubusercontent.com/12413808/42475363-546d9960-83c2-11e8-9b40-bfbbf784eaeb.png)
